pkg/scheduler/queue/tree_queueing_algorithms.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package queue
+
+// DequeueAlgorithm represents the set of operations specific to different approaches to dequeuing. It is applied
+// at the layer-level -- every Node at the same depth in a TreeQueue shares the same DequeueAlgorithm, including state
+// that may be stored in a struct that implements DequeueAlgorithm.
+type DequeueAlgorithm interface {
+	addChildNode(*Node, *Node)
+	deleteChildNode(*Node, *Node) bool
+	dequeueGetNode(*Node) (*Node, bool)
```

Comment:
naming nit: the signature `dequeueGetNode(*Node) (*Node, bool)` is too vague particularly because it takes a node and returns a node - As a consumer/implementer I can't tell is it supposed to return same node as it received as an arg?

`dequeueSelectChildNode`, `dequeueSelectNode`, `dequeueSelectChild`, etc. etc.

Reply:
I think `dequeueSelectNode` is the best; sometimes the (currently-named) `dequeueGetNode` _does_ return itself, so including `Child` in the name might also be misleading.